tools/build_langserver/langserver/analyzer_test.go

```diff
@@ -0,0 +1,99 @@
+package langserver
+
+import (
+	"core"
+	"context"
+	"path"
+	"testing"
+
+	"parse/asp"
+	"tools/build_langserver/lsp"
+
+	"github.com/stretchr/testify/assert"
+	"fmt"
+)
+
+func TestNewAnalyzer(t *testing.T) {
+	a := newAnalyzer()
+	assert.NotEqual(t, nil, a.BuiltIns)
+}
+
+func TestGetStatementsFromFile(t *testing.T) {
+	a := newAnalyzer()
+	core.FindRepoRoot()
+	ctx := context.Background()
+	filepath := path.Join(core.RepoRoot, "tools/build_langserver/langserver/BUILD")
+	uri := lsp.DocumentURI("file://" + filepath)
+	fileContent, err := ReadFile(ctx, uri)
+
+	stmt, err := a.IdentFromFile(uri, fileContent)
+	fmt.Println(err)
+	fmt.Println("Luna", stmt[0].Action.Call.Arguments[0].Value.Val)
+	fmt.Println(stmt)
+
+}
+
+func TestNewRuleDef(t *testing.T) {
+	a := newAnalyzer()
+	// Test header the definition for build_rule
+	buildRule := a.parser.GetAllBuiltinStatements()[0].FuncDef
```

Comment:
This is a bit fragile... there's no real guarantee that `build_rule` is the first builtin (or that `startswith` is the tenth). This also breaks if we change its definition (although I guess they are all subject to that).

Reply:
Yeah, the test is a mess, I intent to change that later :P